app/src/androidTest/java/com/github/se/project/ui/map/LocationPickerTest.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.se.project.ui.map
+
+import MapPickerBox
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class MapPickerBoxTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Test
+  fun mapPickerBox_withInitialLocation_displaysCorrectly_andConfirmButtonTriggersCallback() {
+    // Given initial non-zero coordinates for the test
+    val initialLatitude = 46.5
+    val initialLongitude = 6.6
+    var selectedLocation: Pair<Double, Double>? = null
+
+    // Set the content with MapPickerBox
+    composeTestRule.setContent {
+      MapPickerBox(
+          initialLocation = initialLatitude to initialLongitude,
+          onLocationSelected = { location -> selectedLocation = location })
+    }
+
+    // Check if the map itself is displayed
+    composeTestRule.onNodeWithTag("googleMap").assertIsDisplayed()
+
+    // Check if the helper text displays the correct message
+    composeTestRule.onNodeWithTag("helperText").assertIsDisplayed()
+
+    // Check if the confirm button is displayed and enabled
+    composeTestRule.onNodeWithTag("confirmLocation").assertIsDisplayed().assertIsEnabled()
+
+    // Tap the confirm button
+    composeTestRule.onNodeWithTag("confirmLocation").performClick()
+
+    // Verify the callback was triggered with the correct location
+    assertEquals(initialLatitude to initialLongitude, selectedLocation)
```

Comment:
It is really great that you add a point to check if the callback is correctly triggered!